src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java

```diff
@@ -0,0 +1,174 @@
+ */
+package org.hawkular.client.android.util;
+
+import java.net.URI;
+import java.net.URL;
+
+import org.hawkular.client.android.activity.LoginActivity;
+import org.jboss.aerogear.android.authorization.AuthzModule;
+import org.jboss.aerogear.android.core.Callback;
+import org.jboss.aerogear.android.pipe.http.HeaderAndBody;
+import org.jboss.aerogear.android.pipe.http.HttpException;
+import org.jboss.aerogear.android.pipe.http.HttpProvider;
+import org.jboss.aerogear.android.pipe.http.HttpRestProvider;
+import org.jboss.aerogear.android.pipe.module.AuthorizationFields;
+import org.jboss.aerogear.android.pipe.module.ModuleFields;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Base64;
+import android.util.Log;
+
+public class ModifiedAuthzModule implements AuthzModule {
+
+    Context applicationContext;
+
+    public ModifiedAuthzModule(Context context) {
+        applicationContext = context;
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+
+        if (Preferences.of(applicationContext).exist().equals("true")) {
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public void requestAccess(final Activity activity, final Callback<String> callback) {
+
+        Exception exception;
+        String json;
+        if (Preferences.of(activity).exist().get().equals("true")) {
+            final String key = Preferences.of(activity).key().get();
+            final String secret = Preferences.of(activity).secret().get();
+            JSONObject obj = new JSONObject();
+
+            try {
+                obj.put("key", key);
+                obj.put("secret", secret);
+                callback.onSuccess(obj.toString());
+            } catch (JSONException e) {
+                callback.onFailure(e);
+            }
+        } else {
+
+            AsyncTask<Void, Void, HeaderAndBody> task = new AsyncTask<Void, Void, HeaderAndBody>() {
+                private Exception exception;
+                String json;
+
+                @Override
+                protected HeaderAndBody doInBackground(Void... params) {
+                    HeaderAndBody result;
+                    try {
+                        HttpProvider provider =
+                                new HttpRestProvider(new URL(((LoginActivity) (activity)).backendUrl.toString()
+                                        + "/secret-store/v1/tokens/create"));
+                        provider.setDefaultHeader("Accept", "application/json");
+                        provider.setDefaultHeader("Authorization", "Basic "
+                                + buildLoginData(((LoginActivity) (activity)).username,
+                                ((LoginActivity) (activity)).password));
+                        result = provider.post("a");
```

Comment:
"a"? You don't need to post anything to this endpoint. It seems you can just pass `null` to `HttpRestProvider#post()`